cli/initiator/initiator.go

```diff
@@ -30,76 +28,76 @@ var StartDKG = &cobra.Command{
 		if err != nil {
 			logger.Fatal("😥 Failed to load participants: ", zap.Error(err))
 		}
-		opMap, err := cli_utils.LoadOperators()
+		opMap, err := cli_utils.LoadOperators(logger)
 		if err != nil {
 			logger.Fatal("😥 Failed to load operators: ", zap.Error(err))
 		}
 		logger.Info("🔑 opening initiator RSA private key file")
-		privateKey, encryptedRSAJSON, err := cli_utils.LoadRSAPrivKey()
+		privateKey, err := cli_utils.LoadInitiatorRSAPrivKey(cli_utils.GenerateInitiatorKeyIfNotExisting)
 		if err != nil {
 			logger.Fatal("😥 Failed to load private key: ", zap.Error(err))
 		}
-		dkgInitiator := initiator.New(privateKey, opMap, logger)
+		dkgInitiator := initiator.New(privateKey, opMap, logger, cmd.Version)
 		ethnetwork := e2m_core.MainNetwork
 		if cli_utils.Network != "now_test_network" {
 			ethnetwork = e2m_core.NetworkFromString(cli_utils.Network)
 		}
-		// create a new ID
-		id := crypto.NewID()
 		// start the ceremony
-		depositData, keyShares, err := dkgInitiator.StartDKG(id, cli_utils.WithdrawAddress.Bytes(), operatorIDs, ethnetwork, cli_utils.OwnerAddress, cli_utils.Nonce)
-		if err != nil {
-			logger.Fatal("😥 Failed to initiate DKG ceremony: ", zap.Error(err))
-		}
-		// Save deposit file
-		logger.Info("🎯  All data is validated.")
-		depositFinalPath := fmt.Sprintf("%s/deposit-%s-%v.json", cli_utils.OutputPath, depositData.PubKey, hex.EncodeToString(id[:]))
-		logger.Info("💾 Writing deposit data json to file", zap.String("path", depositFinalPath))
-		err = utils.WriteJSON(depositFinalPath, []initiator.DepositDataJson{*depositData})
-		if err != nil {
-			logger.Warn("Failed writing deposit data file: ", zap.Error(err))
-		}
-		// Save results
-		logger.Info("💾 Writing keyshares payload to file")
-		err = cli_utils.WriteKeyShares(id, keyShares.Payload.PublicKey, keyShares)
-		if err != nil {
-			logger.Warn("Failed writing keyshares file: ", zap.Error(err))
-		}
-		if cli_utils.PrivKey == "" && cli_utils.GenerateInitiatorKey {
-			rsaKeyPath := fmt.Sprintf("%s/encrypted_private_key-%v.json", cli_utils.OutputPath, depositData.PubKey)
-			err = os.WriteFile(rsaKeyPath, encryptedRSAJSON, 0o644)
-			if err != nil {
-				logger.Fatal("Failed to write encrypted private key to file", zap.Error(err))
-			}
-			if cli_utils.PrivKeyPassword == "" {
-				rsaKeyPasswordPath := fmt.Sprintf("%s/password-%v.txt", cli_utils.OutputPath, depositData.PubKey)
-				password, err := crypto.GenerateSecurePassword()
-				if err != nil {
-					logger.Fatal("Failed to generate secure password", zap.Error(err))
-				}
-				err = os.WriteFile(rsaKeyPasswordPath, []byte(password), 0o644)
-				if err != nil {
-					logger.Fatal("Failed to write encrypted private key to file", zap.Error(err))
+		var wg sync.WaitGroup
+		resultChan := make(chan *Result)
+		for i := 0; i < int(cli_utils.Validators); i++ {
+			wg.Add(1)
+			go func(i int) {
+				defer wg.Done()
+				// create a new ID
```

Comment:
IMO you could achieve the same thing with simpler concurrency patterns and only a channel.